packages/react-reconciler/src/ReactUpdateQueue.new.js

```diff
@@ -260,6 +265,30 @@ export function enqueueUpdate<State>(
   }
 }
 
+export function entangleTransitions(root: FiberRoot, fiber: Fiber, lane: Lane) {
+  const updateQueue = fiber.updateQueue;
+  if (updateQueue === null) {
+    // Only occurs if the fiber has been unmounted.
+    return;
+  }
+
+  const sharedQueue: SharedQueue<mixed> = (updateQueue: any).shared;
+  if (isTransitionLane(lane) && root !== null) {
```

Comment:
Is the `root !== null` check here really necessary? Shouldn't Flow prevent that from ever being the case within this function?

Reply:
Yeah good catch, I copy pasted this from the hooks file where the type wasn't already refined